torch/lib/THC/THCTensorScatterGather.cu

```diff
@@ -92,7 +92,7 @@ __global__ void THCudaTensor_gatherKernel(
                                                           tensor, &tensorOffset,
                                                           src, &srcOffset);
 
-    IndexType indexValue = (IndexType)index.data[indexOffset] - TH_INDEX_BASE;
+    long indexValue = index.data[indexOffset] - TH_INDEX_BASE;
```

Comment:
What's the type of `IndexType`? Doesn't it need to be long long (for ARM devices)?